src/components/Carousel.tsx

```diff
@@ -1,18 +1,170 @@
-import React from 'react';
-import './Carousel.scss';
-
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
-
-export default Carousel;
+import { useMemo, useRef, useState } from 'react';
+import Slider from 'react-slick';
+import { images } from '../utils/images';
+import { carouselLengthSelector } from '../utils/carouselLengthSelector';
+
+export const Carousel: React.FC = () => {
+  const sliderRef = useRef<Slider | null>(null);
+  const [itemWidth, setItemWidth] = useState(130);
+  const [frameSize, setFrameSize] = useState(3);
+  const [step, setStep] = useState(3);
+  const [animationDurarion, setAnimationDurarion] = useState(1000);
+  const [isInfinite, setInfinite] = useState(false);
+
+  const next = () => {
+    if (sliderRef.current) {
+      sliderRef.current.slickNext();
+    }
+  };
+
+  const previous = () => {
+    if (sliderRef.current) {
+      sliderRef.current.slickPrev();
+    }
+  };
+
+  const settings = useMemo(() => {
+    return {
+      className: 'carousel',
+      dots: true,
+      infinite: isInfinite,
+      speed: animationDurarion,
+      slidesToShow: frameSize,
+      slidesToScroll: step,
+      arrows: false,
+    };
+  }, [isInfinite, animationDurarion, frameSize, step]);
+
+  return (
+    <>
+      <div className="carousel">
+        <div className="carousel__header">
+          <h1>
+            Carousel with
+            {` ${frameSize} `}
+            images
+          </h1>
+
+          <div className="carousel__buttons">
+            <button
+              type="button"
+              className="button"
+              onClick={previous}
+            >
+              {'<'}
+            </button>
+
+            <button
+              type="button"
+              className="button"
+              onClick={next}
+            >
+              {'>'}
+            </button>
+          </div>
+
+        </div>
+        <Slider ref={sliderRef} {...settings}>
+          {images.map((item) => (
+            <div>
+              <div className="container">
+                <img style={{ width: itemWidth }} src={item} alt="logo" />
+              </div>
+            </div>
+          ))}
+        </Slider>
+
+      </div>
+      <div className="select">
+        <h4 className="select__title">Chose image width:</h4>
+        <button
+          type="button"
+          onClick={() => {
+            setItemWidth(itemWidth - 1);
```

Comment:
```suggestion
            setItemWidth(prevWidth => prevWidth - 1);
```
To avoid problems with the calculation, the previous value should be taken like this
https://reactjs.org/docs/hooks-reference.html#functional-updates